xwiki-pro-macros-ui/src/main/resources/Confluence/Macros/recently-updated.xml

```diff
@@ -0,0 +1,1881 @@
+  ## Gather types based on wheter they are kept or discarded
+  #foreach ($type in $splittedTypeList)
+    #set ($type = $type.strip())
+    #if ($type.startsWith('-'))
+      #set ($usedTypeList = $typesDiscarded)
+      #set ($type = $!stringtool.substring($type, 1))
+    #else
+      #set ($usedTypeList = $types)
+    #end
+    #if ($type.equals('page'))
+      #set ($discard = $usedTypeList.add('page'))
+    #elseif ($type.equals('blogpost') || $type.equals('news'))
+      #set ($discard = $usedTypeList.add('blogpost'))
+    #elseif ($type.equals('attachment'))
+      #set ($discard = $usedTypeList.add('attachment'))
+    #elseif ($type.equals('comment'))
+      #set ($discard = $usedTypeList.add('comment'))
+    #end
+  #end
+  ## Compute types to return
+  #if ($types.size() == 0)
+    #set ($types = $allTypes)
+  #end
+  #set ($discard = $types.removeAll($typesDiscarded))
+  ## Return
+  #set ($return = $NULL)
+  #setVariable("$return" $types)
+#end
+
+## Authors
+#set ($confluenceAuthorsString = "$!wikimacro.parameters.author.strip()")
+#formatSerializedAuthorList($confluenceAuthorsString, $authors)
+
+## Wiki &amp; Spaces
+#set ($confluenceSpacesString = "$!wikimacro.parameters.spaces.strip()")
+#set ($spaces = $NULL)
+#set ($wiki = $doc.wiki)
+#if (
+  $confluenceSpacesString.equals('')
+  || $confluenceSpacesString.equals('@self')
+)
+  ## Search in current space
+  #set ($spaces = $doc.space)
+#elseif (
+  $confluenceSpacesString.equals('@personal')
+  || $confluenceSpacesString.equals('@global')
+  || $confluenceSpacesString.equals('@favorite')
+  || $confluenceSpacesString.equals('@favourite')
+)
+  ## Search everywhere in current wiki
+  ## `@personal`, `@favorite` and `@favourite` do not have an equivalent in XWiki,
+  ## so make them fallback as `@global`
+  #set ($wiki = $doc.wiki)
+#elseif (
+  $confluenceSpacesString.equals('@all')
+  || $confluenceSpacesString.equals('*')
+)
+  ## Search everywhere in every wikis
+  #set ($wiki = $NULL)
+#else
+  ## Search in specified spaces
+  #formatSerializedList($confluenceSpacesString, $spaces)
+#end
+
+## Tags
+#set ($confluenceLabelString = "$!wikimacro.parameters.labels.strip()")
+#formatSerializedList($confluenceLabelString, $tags)
+
+## Types
+#set ($confluenceTypesString = "$!wikimacro.parameters.types.strip()")
+#parseTypes($confluenceTypesString, $types)
+
+## Limit
+#set ($limit = $!wikimacro.parameters.max)
+
+## Theme
+#set ($availableTheme = ['concise', 'social', 'sidebar'])
+#set ($theme = "$wikimacro.parameters.theme.strip()")
+#if (!$availableTheme.contains($theme))
+ #set ($theme = 'concise')
+#end
+#set ($ICON_BY_TYPE = {
+  "DOCUMENT": 'page',
+  "ATTACHMENT": 'attach',
+  "OBJECT_PROPERTY": 'comment'
+})
+#set ($AUTHORING_TEXT_BY_TYPE = {
+  "DOCUMENT": 'contributed',
+  "ATTACHMENT": 'attached',
+  "OBJECT_PROPERTY": 'commented'
+})
+
+#set ($widthStyle = "style='--width: $wikimacro.parameters.width'")
```

Comment:
This allows XWiki syntax injection and thus remote code execution via the width parameter (not here but down where the parameter is printed.